backend/redis_client.py

```
@@ -0,0 +1,377 @@
+        if not self.connected:
+            return False
+
+        try:
+            # Get current leader
+            current_leader = await self.redis.get("astra:resilience:leader")
+            if current_leader and current_leader.decode() == instance_id:
+                # Renew TTL
+                await self.redis.expire("astra:resilience:leader", ttl)
+                logger.debug(f"Leadership renewed for {instance_id} (TTL: {ttl}s)")
+                return True
+            return False
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)